### PR TITLE
Bump dependencies: pre-commit and python-publish

### PR DESCRIPTION
**Pull request by antonagestam**

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -6,6 +6,6 @@ on:
 
 jobs:
   build-and-publish:
-    uses: less-action/reusables/.github/workflows/python-publish.yaml@v2
+    uses: less-action/reusables/.github/workflows/python-publish.yaml@v6
```
Comment (antonagestam) [inline .github/workflows/release.yaml:9]: This brings validation that the release tag equals the packaged version.

**Comment by antonagestam** [comment]
Ouff, type checking is thoroughly broken, I suspect much of it is due to the fixes of managers in django-stubs.

File: tests/project/models.py [hunk 2]
```diff
@@ -17,9 +17,23 @@ class Simple(Model):
...
 
+if TYPE_CHECKING:
+
+    class ParentQuerySet(ExtendedQuerySet["Parent"]):
+        ...
+
+else:
+
+    class ParentQuerySet(ExtendedQuerySet):
+        ...
```
[the remark this answers is not in the document]
Comment (antonagestam) [reply tests/project/models.py:28]: @flaeppe It does include a generic argument, it just doesn't work on the Django versions prior to introducing `QuerySet.__class_getitem__`.

[the remark this answers is not in the document]
Comment (flaeppe) [reply tests/project/models.py:28]: Ah, right

File: tests/project/models.py [hunk 3]
```diff
@@ -30,16 +44,44 @@ def version(self) -> str:
...
+    class ChildQuerySet(ExtendedQuerySet["Child"]):
+        ...
+
+else:
+
+    class ChildQuerySet(ExtendedQuerySet):
+        ...
+
+
+ChildManager = BaseManager.from_queryset(ChildQuerySet)
```
Comment (flaeppe) [inline tests/project/models.py:58]: A new manager could still be built from the `Manager` type, right?

```suggestion
ChildManager = Manager.from_queryset(ChildQuerySet)
```

File: tests/project/models.py [hunk 4]
```diff
@@ -30,16 +44,44 @@ def version(self) -> str:
...
+    class NodeQuerySet(ExtendedQuerySet["Node"]):
+        ...
+
+else:
+
+    class NodeQuerySet(ExtendedQuerySet):
+        ...
+
+
+NodeManager = BaseManager.from_queryset(NodeQuerySet)
```
Comment (flaeppe) [inline tests/project/models.py:78]: Same as above here.

```suggestion
NodeManager = Manager.from_queryset(NodeQuerySet)
```

**Review by flaeppe — APPROVED** [review]
Just had a minor suggestion, no blocker

**Comment by flaeppe** [comment]
This PR should also supersede #83

**Comment by antonagestam** [comment]
Thanks for the review @flaeppe, I updated to build the managers from `Manager` :)

**Review by flaeppe — APPROVED** [review]
LGTM 👍

**Pull request merged**